ocaml_emulator/riscv_ocaml_sim.ml

```diff
@@ -109,13 +116,15 @@ let check_elf () =
 let write_bytes fl bytes =
   let i = ref 0 in
   while !i < Bytes.length bytes do
-    let s = Printf.sprintf "%02x%02x%02x%02x\n"
-	(int_of_char (Bytes.get bytes (!i+3)))
-	(int_of_char (Bytes.get bytes (!i+2)))
-	(int_of_char (Bytes.get bytes (!i+1)))
-	(int_of_char (Bytes.get bytes (!i+0))) in
-    output_string fl s;
-    i := !i + 4
+    let j = ref 0 in
+    while !j < !signature_granularity_ref do
```

Comment:
Why not use a `for`-loop here, then you can avoid hand-incrementing the counter `j`

Reply:
Something like

    for j = 0 to !signature_granularity_ref - 1 do
        ...
    done

Reply:
I don't think that OCaml has a `for` loop by increments other than 1 and -1, so the other loop can't be a `for`-loop alas

Reply:
So, I can only change the inner while loop to a for loop, right? I'll do it in the next commit.